cumulus/node/README.md

```diff
@@ -10,36 +10,40 @@ Compile all the binaries:
 $ cargo build --release
 ```
 
-Build a chain spec which will be used for running the embedded primary node.
-
-```bash
-$ ./target/release/subspace-node build-spec --chain=dev --raw --disable-default-bootnode > dev.json
-```
-
 ### Spin up a local testnet
 
 1. Run a primary node.
 
 ```bash
 $ ./target/release/subspace-node --dev -d tmp --log=txpool=trace,gossip::executor=trace
```

Comment:
Do you need to start it separately now BTW?

Reply:
It's a standalone consensus node, which will be started separately.

Reply:
Hm, but we can combine it with executor node for simpler testing, right?

Reply:
Hmm, I don't see what's the benefit of combining it, this standalone primary full node plays the role of the consensus layer which keeps producing primary blocks, then the executor will process each of them to produce secondary blocks, it's not complex and clear regrading the actual work flow.

Reply:
The benefit is that you wouldn't need to build it first, then running one command, looking at node ID and running second command. You could just use `cargo run` and have everything working in one step.

Reply:
Hm, I see... RPC is actually disabled in that case. I'll try to fix that.

Reply:
>The benefit is that you wouldn't need to build it first, then running one command, looking at node ID and running second command. You could just use cargo run and have everything working in one step.

Sorry that I still don't understand how you can run the standalone primary node and executor node with just one command.

I have been used to building everything first in the release mode to save some disk space as I used an old MBP with 512GB only :D.

Reply:
Executor node has perfectly capable primary node in it already (minus RPC endpoint that is currently disabled, but I'm working on re-enabling among other things). I don't see why you can't use it as RPC endpoint for farmer. It already fires slot info notifications, just connect farmer to it and go!

Reply:
Okay, you mean making the embedded primary node able to produce blocks by connecting to some farmer just like the regular node? Not sure it's a better way as in practice the executor node does not have to be a farmer, the embedded primary node should be just transparent, in the future, the embedded primary full node might be eliminated and replaced by a WS client, all the communications between the primary node and secondary node will be achieved by the RPC requests, apart from the benefits mentioned in [this thread](https://github.com/paritytech/cumulus/issues/545), it also makes sense to me due to the executor node can be a bit lightweight as running a primary full node comes with a cost in any way, can be great if that cost can be saved.

Reply:
> the embedded primary node should be just transparent, in the future, the embedded primary full node might be eliminated and replaced by a WS client

As far as I can see it will not. They are tightly coupled and it is no longer primary node embedded in secondary node, it is secondary node as an optional feature in the primary node and I don't see them being separate going forward.

> all the communications between the primary node and secondary node will be achieved by the RPC requests, apart from the benefits mentioned in https://github.com/paritytech/cumulus/issues/545, it also makes sense to me due to the executor node can be a bit lightweight as running a primary full node comes with a cost in any way, can be great if that cost can be saved

We don't have the same problem as Cumulus has where some independent upstream project forces you to update. We also don't have a use case for running multiple "parachain" nodes with one "relay chain" node, we always have 1 to 1 mapping. As to overhead, the overhead of primary chain can be considered negligible comparing to the load executor will have, hence we don't need to worry about it. Also RPC layer introduces a whole bunch of networking-related edge cases, memory copied, serialization/deserialization and other complexities.

I agree it makes a lot of sense for Polkadot+Cumulus, but I don't see that it is relevant for Primary+Secondary chain in Subspace.

Reply:
> I don't see that it is relevant for Primary+Secondary chain in Subspace.

Well, I have to admit that I don't know Primary+Secondary chain in Subspace then :P

Reply:
> I don't see that it is relevant for Primary+Secondary chain in Subspace.

Well, I have to admit that I don't know Primary+Secondary chain in Subspace then :P